### PR TITLE
CED-77: Running tasks from daemon (vs over SSH) 

### PR DESCRIPTION
**Pull request by nravic**

### CODE REVIEW

**Review by bsmithai — COMMENTED** [review]
looks good so far!

**Review by bsmithai — APPROVED** [review]

File: cmd/client-daemon.go [hunk 1]
```diff
@@ -87,9 +78,29 @@ var clientDaemonCmd = &cobra.Command{
 
 		c.logger.Info().Msgf("daemon started at %s", time.Now().Local())
 
+		// create a subscription to NATS commands from the orchestrator first
 		go c.subscribeToCommands(300)
+
+		if pid == 0 {
+			// we retry startJob up to 5 times until it succeeds everytime we get a new serverCommand
+			var task string = c.config.Client.Task
+			for i := 0; i < 5; i++ {
```
Comment (wfoy) [inline cmd/client-daemon.go:87]: what happens if this doesn't succeed within 5 iterations? Is that possible?

File: cmd/client-daemon.go [hunk 2]
```diff
@@ -103,33 +103,87 @@ var clientDaemonCmd = &cobra.Command{
...
+	// so there's no real concerns here about executing the task without proper validation
+
+	// we want the task to run detached so we can checkpoint it
+	cmd := exec.Command("/bin/sh", "-c", task)
+	cmd.SysProcAttr = &syscall.SysProcAttr{
+		Setsid: true,
+	}
+
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
```
Comment (bsmithai) [inline cmd/client-daemon.go:149]: Do we not need to throw stdin & stdout into /dev/null?

Comment (nravic) [reply cmd/client-daemon.go:149]: yes we do, good catch

Comment (nravic) [reply cmd/client-daemon.go]: fixed!

**Review by bsmithai — APPROVED** [review]

**Comment by bsmithai** [comment]
Amazing!

**Pull request merged**